controller/kv/helper.go

```diff
@@ -2872,40 +2883,145 @@ func (m clusterHelper) PutImportTask(importTask *share.CLUSImportTask) error {
 }
 
 func (m clusterHelper) GetApikeyRev(name string, acc *access.AccessControl) (*share.CLUSApikey, uint64, error) {
-    key := share.CLUSApikeyKey(url.QueryEscape(name))
-    if value, rev, _ := m.get(key); value != nil {
-        var apikey share.CLUSApikey
-        json.Unmarshal(value, &apikey)
-        if !acc.Authorize(&apikey, nil) {    
-            return nil, 0, common.ErrObjectAccessDenied
-        }
-        return &apikey, rev, nil
-    }
-    return nil, 0, common.ErrObjectNotFound
+	key := share.CLUSApikeyKey(url.QueryEscape(name))
+	if value, rev, _ := m.get(key); value != nil {
+		var apikey share.CLUSApikey
+		json.Unmarshal(value, &apikey)
+		if !acc.Authorize(&apikey, nil) {
+			return nil, 0, common.ErrObjectAccessDenied
+		}
+		return &apikey, rev, nil
+	}
+	return nil, 0, common.ErrObjectNotFound
 }
 
 func (m clusterHelper) CreateApikey(apikey *share.CLUSApikey) error {
-    key := share.CLUSApikeyKey(url.QueryEscape(apikey.Name))
-    value, _ := json.Marshal(apikey)
-    // secret_key is already hashed
-    return cluster.PutIfNotExist(key, value, false)
+	key := share.CLUSApikeyKey(url.QueryEscape(apikey.Name))
+	value, _ := json.Marshal(apikey)
+	// secret_key is already hashed
+	return cluster.PutIfNotExist(key, value, false)
 }
 
 // caller needs to decide whether to authorize accessing each returned apikey object
 func (m clusterHelper) GetAllApikeysNoAuth() map[string]*share.CLUSApikey {
-    apikeys := make(map[string]*share.CLUSApikey)
-    keys, _ := cluster.GetStoreKeys(share.CLUSConfigApikeyStore)
-    for _, key := range keys {
-        if value, _, _ := m.get(key); value != nil {
-            var apikey share.CLUSApikey
-            json.Unmarshal(value, &apikey)
-            apikeys[apikey.Name] = &apikey
-        }
-    }
-    return apikeys
+	apikeys := make(map[string]*share.CLUSApikey)
+	keys, _ := cluster.GetStoreKeys(share.CLUSConfigApikeyStore)
+	for _, key := range keys {
+		if value, _, _ := m.get(key); value != nil {
+			var apikey share.CLUSApikey
+			json.Unmarshal(value, &apikey)
+			apikeys[apikey.Name] = &apikey
+		}
+	}
+	return apikeys
 }
 
 func (m clusterHelper) DeleteApikey(name string) error {
 	key := share.CLUSApikeyKey(url.QueryEscape(name))
 	return cluster.Delete(key)
 }
+
+// sigstore
+func (m clusterHelper) PutSigstoreRootOfTrust(rootOfTrust *share.CLUSSigstoreRootOfTrust) error {
+	value, err := enc.Marshal(rootOfTrust)
+	if err != nil {
+		return err
+	}
+	return cluster.Put(share.CLUSSigstoreRootOfTrustKey(rootOfTrust.Name), value)
+}
+
+func (m clusterHelper) GetSigstoreRootOfTrust(rootName string) (*share.CLUSSigstoreRootOfTrust, error) {
+	rootOfTrustData, _, err := m.get(share.CLUSSigstoreRootOfTrustKey(rootName))
+	if err != nil || rootOfTrustData == nil {
+		return nil, err
+	}
+	rootOfTrust := &share.CLUSSigstoreRootOfTrust{}
+	err = json.Unmarshal(rootOfTrustData, rootOfTrust)
```

Comment:
in line 2916 enc.Marshal() is called but here json.Unmarshal() is called & it will always get incorrect json object.